extension/ts/objects/bookForm.ts

```diff
@@ -59,10 +59,22 @@ const insertFormData = (saveData: FormData) => getFormElements()
 				element.value = value;
 				element.checked = checked;
 				element.dispatchEvent(new Event("change"));
+				ensureVisible(element);
 			}
 		}
 	});
 
+const ensureVisible = (element: Element) => {
+	// LibraryThing collections checkboxes are sometimes hidden beneath a div that is not visible
+	// THIS IS BRITTLE and relies on the specific markup tree of LibraryThing
+	if (element.id.startsWith(COLLECTIONS_ID_PREFIX)) {
+		const greatGrandParent = element?.parentElement?.parentElement?.parentElement;
+		if (greatGrandParent?.style?.display === "none") {
+			greatGrandParent.style.display = "";
+		}
+	}
+};
+
```

Comment:
oh this is DISGUSTING but hey whatchagonnado

Reply:
actually idk is there a better way? i don't really know frontend lol

I could a selector from the top but it's! technically just as brittle :(

Reply:
idk either lmao, think it's just necessarily brittle 🤷

Reply:
so the alternative is `element.closest(query)`, but it's just as brittle.

The code might look a little cleaner though